zerver/webhooks/honeybadger/view.py

```diff
@@ -0,0 +1,139 @@
+
+from zerver.decorator import api_key_only_webhook_view, has_request_variables
+from zerver.lib.webhooks.common import check_send_webhook_message, UnexpectedWebhookEventType
+from zerver.lib.request import REQ
+from zerver.lib.response import json_success
+from zerver.models import UserProfile
+
+import re
+
+RATE_EXCEEDED_TEMPLATE = "{message}. For more info: {url}"
+ERROR_RESOLVED_TEMPLATE = "The error {id} was resolved by {actor}. For more info: {url}"
+ERROR_OCCURRED_TEMPLATE = "{klass}: {message} The error({id}) occurred at: {last_notice}. For more info: {" \
+                          "url} "
+COMMENT_TEMPLATE = "On error id: {id} {author} commented \n```quote\n{body}\n```\nFor more info: {url}"
+CHECK_IN_TEMPLATE = "{message}. {name} was expected to report at: {expected_at} and reported at: " \
+                    "{reported_at}. For more info: {url}"
+ASSIGNED_TEMPLATE = "{message}. Error with id: {id} has been assigned to {name}. For more info: {url}"
+UP_AND_DOWN_TEMPLATE = "{message} For more info: {url}"
+
+
+def get_topic(message: str) -> str:
+    t = re.match(r'([[\w\W]*])', message)
+    if not t:
+        raise AssertionError("Unrecognised payload format")
+    topic = t.groups()[0][1:-1]
+    return topic
+
+
+def get_rate_exceeded_topic_and_message(payload: Dict[str, Any]) -> Tuple[str, str]:
+    topic = payload["project"]["name"]
+    topic += "/" + payload["project"]["environments"][0]["name"]
+
+    message = RATE_EXCEEDED_TEMPLATE.format(message=re.sub(r'([[\w\W]*] )', '', payload['message']),
+                                            url=payload['notice']['url'])
+
+    return topic, message
+
+
+def get_resolved_topic_and_message(payload: Dict[str, Any]) -> Tuple[str, str]:
+    topic = get_topic(message=payload["message"])
+
+    message = ERROR_RESOLVED_TEMPLATE.format(id=str(payload['fault']['id']),
+                                             actor=payload['actor']['name'],
+                                             url=payload['fault']['url'])
+
+    return topic, message
+
+
+def get_occurred_topic_and_message(payload: Dict[str, Any]) -> Tuple[str, str]:
+    topic = payload["project"]["name"]
+    topic += "/" + payload["project"]["environments"][0]["name"]
+
+    message = ERROR_OCCURRED_TEMPLATE.format(klass=payload['fault']['klass'],
+                                             message=payload['fault']['message'],
+                                             id=payload['fault']['id'],
+                                             last_notice=payload['fault']['last_notice_at'],
+                                             url=payload['fault']['url'])
+
+    return topic, message
+
+
+def get_down_and_up_topic_and_message(event: str, payload: Dict[str, Any]) -> Tuple[str, str]:
+    topic = payload["project"]["name"]
+    topic += "/" + payload["project"]["environments"][0]["name"]
+
+    message = UP_AND_DOWN_TEMPLATE.format(message=re.sub(r'([[\w\W]*] )', '', payload['message']),
```

Comment:
with the expanded regex in `get_topic` we can get rid this `re.sub`.

Reply:
The payload["message"] need not always contain the full message which is evident from commented.json fixture.